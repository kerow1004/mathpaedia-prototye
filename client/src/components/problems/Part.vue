<template>
  <div v-if="part">
    <div v-if="part.sentence" class="part-sentence">
      <strong>{{ index + 1 }}</strong>. {{ part.sentence }}
    </div>
    <div v-if="part.picture" class="picture-container">
      <img :src="part.picture" class="picture">
    </div>
    <div v-if="part.choices && part.choices.length > 1">
      <ol>
        <li v-for="(choice, index) in part.choices" :key="index">{{ choice }}</li>
      </ol>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Part',
  props: ['part', 'index'],
};
</script>

<style scoped>
  .part-sentence {
    padding: 10px;
    white-space: pre-wrap;
  }

  .picture-container {
    text-align: center;
  }

  .picture {
    width: 200px;
  }
</style>
