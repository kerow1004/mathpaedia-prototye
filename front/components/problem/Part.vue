<template>
  <v-card v-if="part">
    <v-container>
      <v-layout column>
        <v-flex>
          <div v-if="part.sentence" class="part-sentence">
            <strong>{{ index + 1 }}</strong>. {{ part.sentence }}
          </div>
        </v-flex>
        <v-flex>
          <div v-if="part.picture" class="picture-container">
            <img :src="part.picture" class="picture">
          </div>
        </v-flex>
        <v-flex>
          <div v-if="part.choices && part.choices.length > 1" class="part-choices">
            <ol>
              <li v-for="(choice, index) in part.choices" :key="index">{{ choice }}</li>
            </ol>
          </div>
        </v-flex>
      </v-layout>
    </v-container>
  </v-card>
</template>

<script>
export default {
  name: 'Part',
  props: ['part', 'index'],
  mounted() {
    renderMathInElement(this.$el, {
      delimiters: [
        { left: '$$', right: '$$', display: true },
        { left: '\\[', right: '\\]', display: true },
        { left: '$', right: '$', display: false },
        { left: '\\(', right: '\\)', display: false },
      ],
    });
  },
};
</script>

<style scoped>
  .part-sentence {
    white-space: pre-wrap;
  }

  .picture-container {
    text-align: center;
  }

  .picture {
    width: 200px;
  }

  .part-choices {
    margin: 20px;
  }
</style>
