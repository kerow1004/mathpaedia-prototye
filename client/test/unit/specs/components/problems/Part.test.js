import { mount } from '@vue/test-utils';
import Part from '@/components/problems/Part';

describe('Math Part.vue', () => {
  test('renders a part of math problem', () => {
    const sentence = 'math part sentence';
    const picture = 'http://picture.math.com/123456';
    const choices = ['1', '2', '3', '4', '5'];
    const index = 10;

    const wrapper = mount(Part, {
      propsData: {
        part: {
          sentence,
          picture,
          choices,
        },
        index,
      },
    });

    // NOTE: Example test code without snapshot
    // expect(wrapper.find('.part-sentence').exists()).toBe(true);
    // expect(wrapper.find('.part-sentence').text()).toBe(`${index + 1}. ${sentence}`);
    // expect(wrapper.find('.picture-container > .picture').exists()).toBe(true);
    // expect(wrapper.find('.picture-container > .picture').attributes().src).toBe(picture);
    // expect(wrapper.findAll('li')).toHaveLength(5);
    // expect(wrapper.find('li').text()).toBe('1');

    // NOTE: Test code with snapshot
    expect(wrapper.element).toMatchSnapshot();
  });
});
