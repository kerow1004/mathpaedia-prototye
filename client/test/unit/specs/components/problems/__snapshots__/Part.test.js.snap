// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Math Part.vue renders a part of math problem 1`] = `
<div>
  <div
    class="part-sentence"
  >
    <strong>
      11
    </strong>
    . math part sentence
  
  </div>
   
  <div
    class="picture-container"
  >
    <img
      class="picture"
      src="http://picture.math.com/123456"
    />
  </div>
   
  <div>
    <ol>
      <li>
        1
      </li>
      <li>
        2
      </li>
      <li>
        3
      </li>
      <li>
        4
      </li>
      <li>
        5
      </li>
    </ol>
  </div>
</div>
`;
